Allow the default log location to be configurable (e.g. /var/log),
and use separate logs for each display instance (e.g. XWin.0.log).

FIXME: g_pszLogFile should be (const char*) per os/log.c:LogInit().

---
 configure.ac             |    1 +
 hw/xwin/InitOutput.c     |   17 +++++++----------
 hw/xwin/winerror.c       |    7 +++++++
 hw/xwin/winglobals.c     |    2 +-
 include/xwin-config.h.in |    5 ++++-
 5 files changed, 20 insertions(+), 12 deletions(-)

--- origsrc/xorg-server-1.7.0.901/configure.ac	2009-10-13 22:50:18.408257800 -0500
+++ src/xorg-server-1.7.0.901/configure.ac	2009-10-13 22:51:26.268376100 -0500
@@ -1709,6 +1709,7 @@ AC_MSG_RESULT([$XWIN])
 
 if test "x$XWIN" = xyes; then
 	PKG_CHECK_EXISTS($WINDOWSWMPROTO, [WINDOWSWM=yes], [WINDOWSWM=no])
+	AC_DEFINE_DIR(DEFAULT_LOGDIR, logdir, [Default log location])
 	AC_DEFINE_UNQUOTED(XORG_VERSION_CURRENT, [$VENDOR_RELEASE], [Current Xorg version])
 	AC_DEFINE_UNQUOTED(__VENDORDWEBSUPPORT__, ["$VENDOR_WEB"], [Vendor web address for support])
 	AC_CHECK_TOOL(WINDRES, windres)
--- origsrc/xorg-server-1.7.0.901/hw/xwin/InitOutput.c	2009-10-13 22:50:13.868649900 -0500
+++ src/xorg-server-1.7.0.901/hw/xwin/InitOutput.c	2009-10-13 22:52:06.344846000 -0500
@@ -243,7 +243,7 @@ ddxGiveUp (void)
 #endif
 
   if (!g_fLogInited) {
-    LogInit (g_pszLogFile, NULL);
+    g_pszLogFile = LogInit (g_pszLogFile, NULL);
     g_fLogInited = TRUE;
   }  
   LogClose ();
@@ -687,9 +687,6 @@ OsVendorInit (void)
   /* Re-initialize global variables on server reset */
   winInitializeGlobals ();
 
-  LogInit (NULL, NULL);
-  LogSetParameter (XLOG_VERBOSITY, g_iLogVerbose);
-
   winFixupPaths();
 
 #ifdef DDXOSVERRORF
@@ -704,7 +701,7 @@ OsVendorInit (void)
      * avoid the second call 
      */  
     g_fLogInited = TRUE;
-    LogInit (g_pszLogFile, NULL);
+    g_pszLogFile = LogInit (g_pszLogFile, NULL);
   } 
   LogSetParameter (XLOG_FLUSH, 1);
   LogSetParameter (XLOG_VERBOSITY, g_iLogVerbose);
@@ -875,7 +872,7 @@ winUseMsg (void)
 	  "\tEquivalent to XKBOptions in XF86Config files.\n");
 
   ErrorF ("-logfile filename\n"
-	  "\tWrite logmessages to <filename> instead of /tmp/Xwin.log.\n");
+	  "\tWrite logmessages to <filename>.\n");
 
   ErrorF ("-logverbose verbosity\n"
 	  "\tSet the verbosity of logmessages. [NOTE: Only a few messages\n"
@@ -905,7 +902,7 @@ ddxUseMsg(void)
 
   /* Log file will not be opened for UseMsg unless we open it now */
   if (!g_fLogInited) {
-    LogInit (g_pszLogFile, NULL);
+    g_pszLogFile = LogInit (g_pszLogFile, NULL);
     g_fLogInited = TRUE;
   }  
   LogClose ();
@@ -913,9 +910,9 @@ ddxUseMsg(void)
   /* Notify user where UseMsg text can be found.*/
   if (!g_fNoHelpMessageBox)
     winMessageBoxF ("The " PROJECT_NAME " help text has been printed to "
-		  "/tmp/XWin.log.\n"
-		  "Please open /tmp/XWin.log to read the help text.\n",
-		  MB_ICONINFORMATION);
+		  "%s.\n"
+		  "Please open %s to read the help text.\n",
+		  MB_ICONINFORMATION, g_pszLogFile, g_pszLogFile);
 }
 
 /* See Porting Layer Definition - p. 20 */
--- origsrc/xorg-server-1.7.0.901/hw/xwin/winerror.c	2009-10-13 22:50:12.963848300 -0500
+++ src/xorg-server-1.7.0.901/hw/xwin/winerror.c	2009-10-13 22:52:06.360446100 -0500
@@ -44,6 +44,7 @@ extern char *		g_pszCommandLine;
 extern char *		g_pszLogFile;
 extern Bool		g_fSilentFatalError;
 extern Bool		g_fSilentDupError;
+extern Bool		g_fLogInited;
 
 
 #ifdef DDXOSVERRORF
@@ -104,6 +105,12 @@ OsVendorFatalError (void)
   if (g_fSilentFatalError)
     return;
 
+  if (!g_fLogInited) {
+    g_fLogInited = TRUE;
+    g_pszLogFile = LogInit (g_pszLogFile, NULL);
+  }
+  LogClose ();
+
   winMessageBoxF (
           "A fatal error has occurred and " PROJECT_NAME " will now exit.\n" \
 		  "Please open %s for more information.\n",
--- origsrc/xorg-server-1.7.0.901/hw/xwin/winglobals.c	2009-10-13 22:50:09.407042100 -0500
+++ src/xorg-server-1.7.0.901/hw/xwin/winglobals.c	2009-10-13 22:52:06.344846000 -0500
@@ -69,7 +69,7 @@ Bool           g_fAuthEnabled = FALSE;
 HICON		g_hIconX = NULL;
 HICON		g_hSmallIconX = NULL;
 #ifndef RELOCATE_PROJECTROOT
-char *		g_pszLogFile = "/tmp/XWin.log";
+char *		g_pszLogFile = DEFAULT_LOGDIR "/XWin.%s.log";
 #else
 char *		g_pszLogFile = "XWin.log";
 Bool		g_fLogFileChanged = FALSE;
--- origsrc/xorg-server-1.7.0.901/include/xwin-config.h.in	2009-10-13 22:50:18.439457900 -0500
+++ src/xorg-server-1.7.0.901/include/xwin-config.h.in	2009-10-13 22:52:06.360446100 -0500
@@ -24,4 +24,7 @@
 #undef NEED_SNPRINTF
 
 /* Vendor web address for support */
-#undef __VENDORDWEBSUPPORT__
\ No newline at end of file
+#undef __VENDORDWEBSUPPORT__
+
+/* Default log location */
+#undef DEFAULT_LOGDIR
